core/src/service/search_service.rs

```diff
@@ -49,4 +68,350 @@ impl RequestContext<'_, '_> {
 
         Ok(results)
     }
+
+    pub fn start_search(&mut self) -> Result<StartSearchInfo, CoreError> {
+        let files_not_deleted: DecryptedFiles =
+            self.get_all_not_deleted_metadata(RepoSource::Local)?;
+
+        let mut files_info = Vec::new();
+
+        for file in files_not_deleted.values() {
+            if file.is_document() {
+                let content =
+                    match DocumentType::from_file_name_using_extension(&file.decrypted_name) {
+                        DocumentType::Text => self
+                            .read_document(self.config, RepoSource::Local, file.id)
+                            .map(|content| String::from(String::from_utf8_lossy(content.as_ref())))
+                            .map(Some)?,
```

Comment:
1. Why `as_ref` instead of just borrowing?
2. `.to_string()` on the end is more concise (imo) than using `String::from`.
3. Why map the `Some` in a separate call instead of doing it all in one shot?
4. Are you cool with changing `content` to `bytes`?

The above would look like the following. Feel free to commit if you agree.

```suggestion
                            .map(|bytes| Some(String::from_utf8_lossy(&bytes).to_string()))?,
```